Fix interruption gaps and race conditions in Queue and FiberRuntime

NripeshN opened the pull request:
Addresses a race condition in Queue.take where items could be lost if a take is interrupted after the promise is completed. Ensures that items are not lost by putting them back in the queue if necessary. In FiberRuntime, fixes an interruption gap in uninterruptibleMask with synchronous async completion, preserving interruption status and preventing new interruption opportunities on successful effects. Adds tests to verify these behaviors.

/claim #9973

NripeshN committed:
Fix interruption gaps and race conditions in Queue and FiberRuntime

Addresses a race condition in Queue.take where items could be lost if a take is interrupted after the promise is completed. Ensures that items are not lost by putting them back in the queue if necessary. In FiberRuntime, fixes an interruption gap in uninterruptibleMask with synchronous async completion, preserving interruption status and preventing new interruption opportunities on successful effects. Adds tests to verify these behaviors.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/zio/zio?pullRequest=10023) <br/>All committers have signed the CLA.

NripeshN committed:
Adjust test repetition and timeout in QueueSpec and ZIOSpec

NripeshN committed:
Fix tests

NripeshN committed:
Improve test

NripeshN committed:
Remove unwanted test.

NripeshN committed:
lint

NripeshN committed:
Refactor Queue.take and FiberRuntime to improve interruption handling

Simplified the logic in Queue.take by removing unnecessary promise management steps, ensuring that takers are properly handled on interruption. In FiberRuntime, removed the conditional interruption handling for successful effects to eliminate interruption gaps, maintaining consistent behavior during flag reversion.

Abhinanda-Shetty reviewed (commented):
Sorry, But i think this didn't solve the issue!
Race Condition is not minimized or terminated completely!
Items can get disappear.

hearnadam commented:
Does not fix the issue.
